src/ring_theory/polynomial/vieta.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2020 Hanting Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Hanting Zhang
+-/
+import algebra.polynomial.big_operators
+import algebra.big_operators.nat_antidiagonal
+
+/-!
+# Vieta's Formula
+
+The main result is `coeff_of_prod_X_add_C`, which shows that the `k`th coefficient of
+a product of linear terms `X + r i` is the `k`th symmetric polynomial of the constant terms `r i`.
+-/
+
+universes u v w
+open_locale big_operators
+
+open finset polynomial
+
+namespace vieta
+
+variables {α : Type u} [integral_domain α]
+variables {n : ℕ} {r : ℕ → α} {f : polynomial α}
+
+lemma coeff_zero_X_add_C (a : α) : coeff (X + C a) 0 = a :=
```

Comment:
These three lemmas should move to a more general file (probably close after the definition of `coeff_zero_eq_eval_zero`)